omod/src/main/webapp/resources/dhisconnector-runreports.js

```diff
@@ -68,7 +69,12 @@ function onMappingSelect() {
             initializeMonthlyPicker();
             break
         case 'Yearly':
-            initializeYearlyPicker();
+            financialYearMonth = 'January';
+            initializeFinancialYearPicker();
+            break;
+        case 'FinancialJuly':
+            financialYearMonth = 'July';
+            initializeFinancialYearPicker();
```

Comment:
You could check if the periodType starts with 'Financial' and then extract the month and pass it as a parameter. 

It will support all yearly periods

Reply:
what about Yearly type?

Reply:
pass it as january

Reply:
the yearly case will be there